solr/core/src/test-files/solr/solr.xml

```diff
@@ -35,11 +35,6 @@
     <int name="connTimeout">${connTimeout:15000}</int>
   </shardHandlerFactory>
 
-  <transientCoreCacheFactory name="transientCoreCacheFactory" class="TransientSolrCoreCacheFactoryDefault">
```

Comment:
TBH , I do not understand this abstraction.

Why is `transientCoreCacheFactory` an abstration?

We may have a `CoreFactory` or something like that as an abstraction and `TransientSolrCoreCacheFactory` could be one impl

Reply:
TransientCoreCache was pluggable.  I don't think this is the right pluggable thing, so I removed its plug-ability.  Instead, I think "SolrCores" is better suited to be pluggable.  In so doing, the default/base SolrCores doesn't have to know about transient cores, and is thus simpler to understand for nearly everyone.  Businesses like my employer (in the definite minority) can deal with it in a different way than previously.

You propose a CoreFactory?  Today, cores are created in CoreContainer (as you know, I'm sure).  I'm trying to narrowly limit the scope of this PR to not be noticed by code that doesn't care about transient cores.

Reply:
honestly , a `CoreFactory` is a better abstraction. I know it's more work. But, it is better

Reply:
I don't think it's an either-or, I think of this as more evolutionary.  Feel free to propose CoreFactory separately.  You haven't outlined your whole idea but I presume SolrCores would *become* CoreFactory.  Progress.

Reply:
?

This PR removes one "Factory" -- TransientCoreCacheFactory (the thing you just referred to).  I don't think it's needed at all.  But it doesn't remove TransientCoreCache, perhaps what you meant?  Regardless, I think of this thing as more of an implementation detail inside how the new TransientSolrCores works.  If you haven't reviewed this PR, please do before asking more questions.

I suspect your proposal is for there to be some abstraction that is responsible for creating the cores _and_ managing/holding them generally.  I think it makes sense for "SolrCores" to be that abstraction.  Maybe you didn't mean one abstraction for both.  I'm not sure it matters; maybe you should propose something somewhere.

Reply:
I'm seeing a new pluggable component as follows
```
 <solrCores class="org.apache.solr.core.TransientSolrCores">
    <int name="transientCacheSize">4</int>
  </solrCores>
```

The name ` <solrCores>` actually doesn't make any sense. It would need a more descriptive name say `<coreFactory>` , `<coreManager>`  or something like that

Reply:
`<coreManager>` works for me because its a reasonable name today and in the future if this class takes on core creation responsibility.

> The ticket doesn't even say that you are introducing a new type of component , what the configuration looks like or what the responsibilities are for that component type

From the JIRA issue: "This proposal would make SolrCores itself pluggable".  So there is no new type of component, and no change of responsibilities.  Yeah I could have shared what the plug-ability looks like.  I'm thinking this is so expert that we shouldn't really say anything about it in the ref guide.

Reply:
As we are making `SolrCores` pluggable, we need to clearly define the following in JIRA itself:

- the responsibilities of the plugin
- the configuration of the plugin

Reply:
There is no change of responsibilities here.  Agreed on the configuration; will do.

Reply:
> There is no change of responsibilities here.

I didn't say there is a change in responsibilities. If it is not a pluggable component, the responsibilities do not matter and we can change them as often as we wish. Making it a pluggable component means we have to have a clear definition of the contract

Reply:
I'm confused how I should satisfy the "clear definition of the contract", particularly for pre-existing things that are considered "expert" to plug in / not documented in the ref guide.

Reply:
`SolrCores` is a concrete class . I'm against making a concrete class pluggable. Lets start with an interface extracted from `SolrCores`

Reply:
I could imagine your asks of me snow-balling to fully documenting every method.  I'd like to remind the context as to why this JIRA issue exists and its scope.  Today there is too much complexity around this code relating to transient cores -- a very niche feature that that is basically always enabled even though almost nobody configures cores to be transient.  I'm trying to ensure that the typical code-path doesn't involve any transient core stuff, and also not needlessly change APIs that don't need to change.  In other words, do a fairly narrow refactoring that is a positive step forward.  I don't think this should _require_ documenting existing things lacking documentation.  Not even splitting an interface honestly but I'll do it.

Reply:
>  I don't think this should require documenting existing things lacking documentation.

I'm not asking for a ref guide section. The JIRA can have a basic definition of what this plugin means.